src/plugin/file/kml/tour/toursoundcue.js

```diff
@@ -0,0 +1,141 @@
+goog.provide('plugin.file.kml.tour.SoundCue');
+
+goog.require('goog.Promise');
+goog.require('goog.async.nextTick');
+goog.require('plugin.file.kml.tour.AbstractTourPrimitive');
+
+
+/**
+ * Plays an audio file during the tour.
+ * @param {string} href The URL to the audio file.
+ * @param {number=} opt_delayedStart Delay before playing the file.
+ * @extends {plugin.file.kml.tour.AbstractTourPrimitive}
+ * @constructor
+ */
+plugin.file.kml.tour.SoundCue = function(href, opt_delayedStart) {
+  plugin.file.kml.tour.SoundCue.base(this, 'constructor');
+  this.isAsync = true;
+
+  /**
+   * The audio element.
+   * @type {HTMLAudioElement|undefined}
+   * @private
+   */
+  this.audio_ = undefined;
+
+  /**
+   * URL for the audio file.
+   * @type {string}
+   * @private
+   */
+  this.href_ = href;
+
+  /**
+   * Delay before playing the audio file.
+   * @type {number}
+   * @private
+   */
+  this.delayedStart_ = Math.max(opt_delayedStart || 0, 0);
+
+  /**
+   * Remaining duration on the wait.
+   * @type {number|undefined}
+   * @private
+   */
+  this.remaining_ = undefined;
+
+  /**
+   * The last time the wait was started.
+   * @type {number}
+   * @private
+   */
+  this.start_ = 0;
+
+  /**
+   * The active timeout id.
+   * @type {number|undefined}
+   * @private
+   */
+  this.timeoutId_ = undefined;
+};
+goog.inherits(plugin.file.kml.tour.SoundCue, plugin.file.kml.tour.AbstractTourPrimitive);
+
+
+/**
+ * @inheritDoc
+ */
+plugin.file.kml.tour.SoundCue.prototype.execute = function() {
+  var interval = this.getInterval();
+  if (!this.audio_ && interval > 0) {
+    this.timeoutId_ = setTimeout(this.playAudio_.bind(this), interval);
+    this.start_ = Date.now();
+  } else {
+    this.playAudio_();
+  }
+
+  return goog.Promise.resolve();
+};
+
+
+/**
+ * Play the audio file.
+ * @private
+ */
+plugin.file.kml.tour.SoundCue.prototype.playAudio_ = function() {
```

Comment:
Could this just use the existing AudioManager support? Or could AudioManager be extended instead?

Reply:
The AudioManager is for shared sounds in the application and is intended to be used for notifications. We use it in external plugins to notify things like data arriving from a streaming service. KML SoundCue's typically point to music or voice overs that relate to the tour and didn't seem to fit with the AudioManager.

Reply:
I simplified SoundCue quite a bit by adding asynchronous execution support to `plugin.file.kml.tour.Wait` and extending that for the timeout behavior. Now it simply manages the audio clip and leaves the rest to Wait.

Reply:
OK.

Reply:
That's fine that the sounds are not added, but I'd expect it to still obey the overall mute setting in there.

Reply:
Thanks I hadn't considered that. Fixed and tests updated.